unity-renderer/Assets/DCLServices/PlacesAPIService/PlacesAPIService.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            this.client = client;
+        }
+        public void Initialize() { }
+
+        public async UniTask<(IReadOnlyList<IHotScenesController.PlaceInfo> places, int total)> GetMostActivePlaces(int pageNumber, int pageSize, CancellationToken ct, bool renewCache = false)
+        {
+            var createNewPointer = false;
+
+            if (!activePlacesPagePointers.TryGetValue(pageSize, out var pagePointer)) { createNewPointer = true; }
+            else if (renewCache)
+            {
+                pagePointer.Dispose();
+                activePlacesPagePointers.Remove(pageSize);
+                createNewPointer = true;
+            }
+
+            if (createNewPointer)
+            {
+                activePlacesPagePointers[pageSize] = pagePointer = new LambdaResponsePagePointer<IHotScenesController.PlacesAPIResponse>(
+                    $"", // not needed, the consumer will compose the URL
+                    pageSize, disposeCts.Token, this, TimeSpan.FromSeconds(30));
+            }
+
+            (IHotScenesController.PlacesAPIResponse response, bool _) = await pagePointer.GetPageAsync(pageNumber, ct);
+
+            foreach (IHotScenesController.PlaceInfo place in response.data)
+            {
+                CachePlace(place);
+            }
+
+            return (response.data, response.total);
+        }
+
+        public async UniTask<IHotScenesController.PlaceInfo> GetPlace(Vector2Int coords, CancellationToken ct, bool renewCache = false)
+        {
+            if (renewCache)
+                placesByCoords.Remove(coords);
+            else if (placesByCoords.TryGetValue(coords, out var placeInfo))
+                return placeInfo;
+
+            var place = await client.GetPlace(coords, ct);
+            CachePlace(place);
+            return place;
+        }
+
+        public async UniTask<IHotScenesController.PlaceInfo> GetPlace(string placeUUID, CancellationToken ct, bool renewCache = false)
+        {
+            if (renewCache)
+                placesById.Remove(placeUUID);
+            else if (placesById.TryGetValue(placeUUID, out var placeInfo))
+                return placeInfo;
+
+            var place = await client.GetPlace(placeUUID, ct);
+            CachePlace(place);
+            return place;
+        }
+
+        public async UniTask<IReadOnlyList<IHotScenesController.PlaceInfo>> GetFavorites(CancellationToken ct, bool renewCache = false)
+        {
+            const int CACHE_EXPIRATION = 30; // Seconds
+
+            // We need to pass the source to avoid conflicts with parallel calls forcing renewCache
+            async UniTask RetrieveFavorites(UniTaskCompletionSource<List<IHotScenesController.PlaceInfo>> source)
+            {
+                // We dont use the ct param, otherwise the whole flow would be cancel if the first call is cancelled
+                var favorites = await client.GetFavorites(disposeCts.Token);
```

Comment:
Are you sure you don't want to `catch` any exceptions here? Especially `OperationCanceled`?